### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by Alfred-KInyua**
### In this project, I managed to  add the following project requirement 

**Splash screen**

A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

**Sign up and log in pages**

The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

**Home page (catagories page)**

When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

**Transactions page**

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.

**"Add a new category" page**

The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.

**"Add a new transaction" page**

The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user clicks (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
Testing requirements
Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,88 @@
...
+> In this project, I created a budget calculator website, which has categories of budgets in which you can enter transaction cost for each sub-group in other to get the total budget price in that particular budget category.
 
-Things you may want to cover:
+## Link to video
 
-* Ruby version
+- [Explain](https://www.loom.com/share/c0e9967ce7c84c6ca09f8257ad4f1dc6)
 
-* System dependencies
+## Link to App
```
Comment (OLIPLICHE) [inline README.md]: - Please deploys the application using Heroku, Netlify, or GitHub Pages.
- The project is deployed properly, the home and detail page works as expected and the link to the deployed project 
must be present in the README.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,88 @@
...
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+> In this project, I created a budget calculator website, which has categories of budgets in which you can enter transaction cost for each sub-group in other to get the total budget price in that particular budget category.
 
-Things you may want to cover:
+## Link to video
 
-* Ruby version
+- [Explain](https://www.loom.com/share/c0e9967ce7c84c6ca09f8257ad4f1dc6)
```
Comment (OLIPLICHE) [inline README.md]: - Please record a new video, presenting the features of the project. Also, highlight a piece of the code you find interesting or are proud of making as you did in the previous video

File: app/views/users/_header.html.erb [hunk 3]
```diff
@@ -23,9 +23,8 @@
     <% else %>
       <div class="collapse navbar-collapse" id="navbarSupportedContent">
         <ul class="navbar-nav me-auto mb-2 mb-lg-0">
-          <li class="nav-item">
-            <p>Hello there!</p>
-          </li> 
+          <li class="nav-item"></li>  
+        
         </ul>
```
Comment (OLIPLICHE) [inline app/views/users/_header.html.erb:28]: Nice Job 👍 
Capture

File: spec/requests/category_spec.rb [hunk 4]
```diff
@@ -4,8 +4,8 @@
   include Devise::Test::IntegrationHelpers
   let(:user) do
     User.create(
-      name: 'John',
-      email: 'john1@gmail.com',
+      name: 'Alfred',
+      email: 'alf@gmail.com',
       password: '1234567'
```
Comment (OLIPLICHE) [inline spec/requests/category_spec.rb:9]: Nice 👍 
Capture

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Alfred-KInyua, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Danieldotcomcoder — CHANGES_REQUESTED** [review]
Hello @Alfred-KInyua , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- Good Video Presentation 👍 
- Professional ReadMe file 👍 

 ## Required Changes ♻️

- The Deployed version is not working properly , when I click signup or login I get an error:
Kindly fix this issue.
image


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @Alfred-KInyua 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**